lib/json_matchers/matcher.rb

```diff
@@ -2,18 +2,28 @@
 
 module JsonMatchers
   class Matcher
+    def self.extract_response_body(response)
+      if response.respond_to?(:body)
+        response.body
+      elsif response.is_a?(String)
+        response
+      else
+        fail 'Response does not have a #body method and is ' \
+          'not a string. It has the class class ' \
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.